client/asset/dcr/rpcwallet.go

```diff
@@ -890,6 +891,16 @@ func (w *rpcWallet) rpcClientRawRequest(ctx context.Context, method string, args
 	return nil
 }
 
+// AbandonTransaction attempts to abandon a mempool transaction.
+// Part of the Wallet interface.
+func (w *rpcWallet) AbandonTransaction(ctx context.Context, hash *chainhash.Hash) error {
+	err := w.rpcClientRawRequest(ctx, methodAbandonTransaction, anylist{hash.String()}, nil)
```

Comment:
Hopefully this won't be a problem:

```
abandontransaction "hash"

Remove an unconfirmed transaction and all dependent transactions
```

Say the redeem created a UXTO that while unconfirmed was then selected to fund a new swap...
Detecting that might be a challenge though.

Reply:
I can remove this if you would like as it is not really part of the issue, I guess it's a separate issue but something I have hit by trying to redeem dust transactions with dcrwallet in spv mode. Then the wallet uses those same inputs for something else. But then it still keeps the bad transaction around in its db.

Or I could be misremembering.

Maybe this is a bug in dcrwallet that it keeps the tx around even after the inputs were spent by something else. I think I will just remove abandoning here, try to discern if it is a bug in dcrwallet, and try to get it fixed there if possible.

Incidentally, the vspd fee payment logic has AbandonTransaction as part of its wallet interface, so may have the same worry there.